src/components/modals/ScreenshotModal/ScreenshotModal.component.jsx

```diff
@@ -8,17 +8,73 @@ import { Copy32Icon, Save24Icon } from '../../../icons';
 import { loginHandler } from '../SignInModal';
 import { useAuthContext } from '../../../contexts';
 import Toolbar from '../../scenegraph/Toolbar';
+import { db, storage } from '../../../services/firebase';
+import html2canvas from 'html2canvas';
+import { getDownloadURL, ref, uploadBytes } from 'firebase/storage';
+import {
+  collection,
+  doc,
+  getDoc,
+  serverTimestamp,
+  updateDoc
+} from 'firebase/firestore';
 
 function ScreenshotModal({ isOpen, onClose }) {
   const storedScreenshot = localStorage.getItem('screenshot');
   const parsedScreenshot = JSON.parse(storedScreenshot);
   const { currentUser } = useAuthContext();
-  const saveScreenshot = (value) => {
+  const saveScreenshot = async (value) => {
     const screenshotEl = document.getElementById('screenshot');
     screenshotEl.play();
     screenshotEl.setAttribute('screentock', 'type', value);
     screenshotEl.setAttribute('screentock', 'takeScreenshot', true);
+    try {
+      await new Promise((resolve) => setTimeout(resolve, 1000));
+      const aframeScene = document.getElementById('screentock-destination');
+      const canvas = await html2canvas(aframeScene);
```

Comment:
- this new `canvas` element is created from an existing image using html2canvas
- I think this is not needed and even if it is, not sure we need a package to do this?
- Instead, see below